code/game/objects/items/devices/flashlight.dm

```diff
@@ -299,4 +301,100 @@ obj/item/flashlight/lamp/bananalamp
 	anchored = TRUE
 	var/range = null
 	unacidable = TRUE
-	burn_state = LAVA_PROOF
\ No newline at end of file
+	burn_state = LAVA_PROOF
+
+/obj/item/flashlight/glowstick
+	name = "green glowstick"
+	desc = "A military-grade glowstick."
+	w_class = WEIGHT_CLASS_SMALL
+	brightness_on = 4
+	color = LIGHT_COLOR_GREEN
+	icon_state = "glowstick"
+	item_state = "glowstick"
+	var/fuel = 0
+
+/obj/item/flashlight/glowstick/Initialize()
+	fuel = rand(1600, 2000)
+	light_color = color
+	. = ..()
+
+/obj/item/flashlight/glowstick/Destroy()
+	processing_objects.Remove(src)
+	. = ..()
+
+/obj/item/flashlight/glowstick/process()
+	fuel = max(fuel - 1, 0)
+	if(!fuel)
+		turn_off()
+		processing_objects.Remove(src)
+		update_icon()
+
+/obj/item/flashlight/glowstick/proc/turn_off()
+	on = FALSE
+	update_icon()
+
+/obj/item/flashlight/glowstick/update_icon()
+	item_state = "glowstick"
+	cut_overlays()
+	if(!fuel)
+		icon_state = "glowstick-empty"
+		cut_overlays()
+		update_brightness(0)
+	else if(on)
+		var/mutable_appearance/glowstick_overlay = mutable_appearance(icon, "glowstick-glow")
+		glowstick_overlay.color = color
+		add_overlay(glowstick_overlay)
+		item_state = "glowstick-on"
+		update_brightness(brightness_on)
+	else
+		icon_state = "glowstick"
+		cut_overlays()
+
+/obj/item/flashlight/glowstick/attack_self(mob/user)
+	if(!fuel)
+		to_chat(user, "<span class='notice'>[src] is spent.</span>")
+		return
+	if(on)
+		to_chat(user, "<span class='notice'>[src] is already lit.</span>")
+		return
+
+ 	. = ..()

```

Comment:
There is a space at the start of this line which meant this was failing during my testing.